mlrun/api/db/sqldb/db.py

```diff
@@ -962,10 +967,19 @@ def store_feature_set(
         if not tag and not uid:
             raise ValueError("cannot store feature set without reference (tag or uid)")
 
+        feature_set_project = feature_set.metadata.project
+        if feature_set_project and feature_set_project != project:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"feature-set object with conflicting project name - {feature_set_project}"
+            )
```

Comment:
Move to function

Reply:
I'm going to change this part anyway in the next PR (adding feature-vectors) which will place a lot of the "common logic" in functions, so I'd prefer to wait with this minor change for the next PR.